hack/sync-ec-cli-tasks.sh

```diff
@@ -26,28 +26,29 @@ set -o nounset
 
 EC_CLI_REPO_PATH="${1}"
 
-cp -r "${EC_CLI_REPO_PATH}/tasks" .
+collect_remotes() {
+  echo "$(git branch --remote --format '%(refname:lstrip=-1)' --sort=refname --list 'origin/release-v*')"
+}
 
-pushd tasks > /dev/null
+# helper function to add tasks to a git branch
+add_tasks() {
+  local branch=${1}
+  local remote_branch=${2}
+  pushd "${EC_CLI_REPO_PATH}" > /dev/null
+  git checkout "${branch}"
+  popd > /dev/null
+  git checkout -B "${branch}" --track "${remote_branch}"
+  cp -r "${EC_CLI_REPO_PATH}/tasks" .
+  diff="$(git diff)"
+  if [[ -z "${diff}" ]]; then
+      echo "No changes to sync"
+      return
+  fi
 
-images="$(grep -r -h -o -w 'quay.io/enterprise-contract/ec-cli:.*' | grep -v '@' | sort -u)"
-
-for image in $images; do
-    echo "Resolving image $image"
-    digest="$(skopeo manifest-digest <(skopeo inspect --raw "docker://${image}"))"
-    pinned_image="${image}@${digest}"
-    echo "↳ ${pinned_image}"
-    find . -type f -exec sed -i "s!${image}!${pinned_image}!g" {} +
-done
```

Comment:
I know this is merged already, but I'm not sure I understand why the image pinning code here got removed. Is that intentional? Was it moved somewhere else maybe?

Reply:
My understanding is that pinning was interesting when we were copying from the `main` branch of ec-cli, now we're copying from the `release-v*` branches and the intent is to have the images pinned already there. Am I way off base @joejstuart?

Reply:
Just saw this! @zregvart you are right, but now I'm not sure the pinning will work until PAC is upgraded. Looking at it now though.